src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,106 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
 
     @Override
     public void add(T value) {
-
+        if (size == elements.length) {
+            ensureCapacity();
+        }
+        elements[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < size) {
+            if (size + 1 >= elements.length) {
+                ensureCapacity();
+            }
+            System.arraycopy(elements, index, elements, index + 1, size - index);
+            elements[index] = value;
+            size++;
+        } else {
+            throw new ArrayIndexOutOfBoundsException();
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        int resultLength = size + list.size();
```

Comment:
you can solve it with just one for loop and call you add() method for each element of input list, it will be much shorter